sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -454,50 +457,87 @@ private synchronized void triggerIngestion(final @NonNull GroupState groupState)
             return;
         }
 
-        /* Get a batch from Persistence. */
-        final List<Log> batch = new ArrayList<>(maxFetch);
-        final int stateSnapshot = mCurrentState;
-        final String batchId = mPersistence.getLogs(groupState.mName, groupState.mPausedTargetKeys, maxFetch, batch, null);
+        /* Get auth token. */
+        List<AuthTokenInfo> authTokenHistory = null;
+        AuthTokenStorage authTokenStorage = AuthTokenContext.getInstance().getStorage();
+        if (authTokenStorage != null) {
+            authTokenHistory = authTokenStorage.getTokenHistory();
+        }
+        if (authTokenHistory == null) {
+            authTokenHistory = Collections.singletonList(new AuthTokenInfo(null, null, null));
+        }
+        ListIterator<AuthTokenInfo> iterator = authTokenHistory.listIterator();
+        while (iterator.hasNext()) {
+            AuthTokenInfo authTokenInfo = iterator.next();
+            final String authToken;
+            Date startTime = null;
+            Date endTime = null;
+            if (authTokenInfo != null) {
+                authToken = authTokenInfo.getAuthToken();
+                startTime = authTokenInfo.getStartTime();
+                endTime = authTokenInfo.getEndTime();
+            } else {
+                authToken = null;
+            }
 
-        /* Decrement counter. */
-        groupState.mPendingLogCount -= maxFetch;
+            /* Delete logs without correct token. */
+            if (!iterator.hasPrevious() && startTime != null) {
+                mPersistence.deleteLogs(startTime);
+            }
 
-        /* Nothing more to do if no logs. */
-        if (batchId == null) {
-            return;
-        }
-        AppCenterLog.debug(LOG_TAG, "ingestLogs(" + groupState.mName + "," + batchId + ") pendingLogCount=" + groupState.mPendingLogCount);
+            /* Get a batch from Persistence. */
+            final List<Log> batch = new ArrayList<>(maxFetch);
+            final int stateSnapshot = mCurrentState;
```

Comment:
```suggestion
            int stateSnapshot = mCurrentState;
```

Reply:
No, it's used inside runOnUiThread block